packages/xrpl/src/Wallet/index.ts

```diff
@@ -197,6 +198,38 @@ class Wallet {
     })
   }
 
+  /**
+   * Derives a wallet from secret numbers.
+   *
+   * @param secretNumbers - A string consisting of 8 times 6 numbers (whitespace delimited) used to derive a wallet.
+   * @param opts - (Optional) Options to derive a Wallet.
+   * @param opts.masterAddress - Include if a Wallet uses a Regular Key Pair. It must be the master address of the account.
+   * @param opts.algorithm - The digital signature algorithm to generate an address for.
+   * @returns A Wallet derived from secret numbers.
+   * @throws ValidationError if unable to derive private key from secret number input.
+   */
+  public static fromSecretNumbers(
+    secretNumbers: string[] | string,
+    opts: { masterAddress?: string; algorithm?: ECDSA } = {},
+  ): Wallet {
+    let numbersArray: string[] = []
+
+    if (typeof secretNumbers === 'string') {
+      numbersArray = Utils.parseSecretString(secretNumbers)
+    } else if (Array.isArray(secretNumbers)) {
+      numbersArray = secretNumbers
+    } else {
+      throw new ValidationError('Unable to parse the given secret numbers.')
```

Comment:
unnecessary use of `let` instead of `const` is a pet peeve of mine lol. you can rewrite this like the following to not require a let:


```javascript
const numbersArray: string[] = []

if (typeof secretNumbers === 'string') {
  numberArray.concat(Utils.parseSecretString(secretNumbers))
} else if (Array.isArray(secretNumbers)) {
  numbersArray.concat(secretNumbers)
} ....
```

Reply:
alternatively, there's a typescript syntax that lets you use `let` but also tell typescript that you expect the variable to never _actually_ be nullish. IMO this is strange to look at, so i prefer the above with `const` but another option:

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

```javascript
let numbersArray!: string []

if (typeof secretNumbers === 'string') {
  numbersArray = Utils.parseSecretString(secretNumbers)
} ....
```

Reply:
used your first solution! :)

Reply:
@nixer89 , first solution will not change original `numbersArray` — `.concat()` does not change the existing arrays, but instead returns a new array. Example with `let` is right.